udkm1Dsim/simulations/magnetization.py

```diff
@@ -1013,6 +1013,18 @@ def convert_cartesian_to_polar(cartesian):
 
         """
         polar = np.zeros_like(cartesian)
+
+        xs=cartesian[..., 0]
+        ys=cartesian[..., 1]
+        zs=cartesian[..., ]
+        trans_amplitudes=np.hypot(xs, ys)
+        amplitudes=np.hypot(trans_amplitudes, zs)
+        zeromag_mask=amplitudes!=0.
+        polar[..., 0]=amplitudes   
+        polar[..., 1]=np.arctan2(ys, xs)               
+        polar[zeromag_mask, 2]=np.arccos(np.divide(zs,amplitudes))
+        polar[~zeromag_mask, 2]=0.
```

Comment:
you do not need that line, since the `polar` array is initialized as zeros

Reply:
Codeline was cut and checked with flake8 before the last commit